File: mpi-reduce.f
       Program reduce
       include 'mpif.h'
       integer size, rank, count,a,a_sum, tag, ierr
       integer stat(MPI_STATUS_SIZE)   
       call MPI_INIT(ierr)
       call MPI_COMM_RANK(MPI_COMM_WORLD, rank, ierr)
       call MPI_COMM_SIZE(MPI_COMM_WORLD, size, ierr)
       a=rank
       write(*,10) 'En proceso ', rank, ' a=',a
   10  Format(1x,A15,2x,I2, A4,I2)
       call MPI_REDUCE(a,a_sum,1,MPI_INTEGER,
     & MPI_SUM,0,MPI_COMM_WORLD,ierr)
       if (rank .eq. 0) then
       write(*,20) 'La suma de a es ',a_sum
   20  Format(1x,A20,2x,I2)
       endif
       call MPI_FINALIZE(ierr)
       end

File: mpi-reduce.c
#include <mpi.h>
#include <stdio.h>
int main(int argc, char** argv){
int size, rank, ierr;
MPI_Init(&argc,&argv);
MPI_Comm_size(MPI_COMM_WORLD,&size);
MPI_Comm_rank(MPI_COMM_WORLD,&rank);
int a, a_sum;
a=rank;
  printf("En proceso %i a = %i \n",rank,a);
     MPI_Reduce(&a, &a_sum,1,MPI_REAL,MPI_SUM,0,MPI_COMM_WORLD);
if( rank == 0 ){
  printf("La reduccion de a es =%i \n",a_sum);}
 ierr = MPI_Finalize();
}
